src/common/pool_map.c

```diff
@@ -408,10 +398,8 @@ pool_buf_attach(struct pool_buf *buf, struct pool_component *comps,
 			buf->pb_target_nr++;
 		else if (comps[0].co_type == PO_COMP_TP_NODE)
 			buf->pb_node_nr++;
-		else if (comps[0].co_type == PO_COMP_TP_RACK)
-			buf->pb_domain_nr++;
 		else
-			D_ASSERTF(0, "invalid type %d\n", comps[0].co_type);
+			buf->pb_domain_nr++;
```

Comment:
I understand why it's like this but now node can be domain as well, so we might want to do some cleanup to avoid confusing.